lib/collection.js

```diff
@@ -1525,7 +1525,7 @@ Collection.prototype.createIndex = function(fieldOrSpec, options, callback) {
   callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;
 
   options = args.length ? args.shift() || {} : {};
-  options = typeof callback === 'function' ? options : callback;
+  options = typeof callback === 'function' || typeof callback === 'undefined' ? options : callback;
   options = options == null ? {} : options;
```

Comment:
These two lines seem unnecessary:
```
options = typeof callback === 'function' || typeof callback === 'undefined' ? options : callback;
options = options == null ? {} : options;
```

In the first case where the function is called with two params (`col.createIndex(index, callback)`):

1. `callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;` puts the callback function in `callback` and makes `args` empty.
2. `options = args.length ? args.shift() || {} : {};` puts `{}` in `options` because `args.length` is `0`.

In the second case where the function is called with three params (`col.createIndex(index, options, callback)`):

1. `callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;` puts the callback function in `callback` and makes `args` have a length of 1 (only `options`).
2. `options = args.length ? args.shift() || {} : {};` puts the second param into `options` because it's the only thing left in `args`.

Reply:
Indeed, there is also the third cases where you call the function with promises (`col.createIndex(index, options)`):
which still works

1. `callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;` sets the callback function as undefined
2. `options = args.length ? args.shift() || {} : {};` puts the second parameter in the `options` because it's  the only thing left in `args`

and fourth case where the function is called with one parameter only (`col.createIndex(index)`):

1. `callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;` sets the callback function as undefined
2. `options = args.length ? args.shift() || {} : {};` puts `{}` in the `options` because `args.length` = `0`

Should I write the unit tests and make the update ?

Thanks for your answer

Reply:
Thanks for your work on this @yehudi! I'm not involved in this project, we just ran into the same bug you did, so I'll defer to @daprahamian on your question.

Reply:
@yehudi yes, all four cases are currently valid. Please write unit tests to confirm that you no longer experience your error.

Reply:
Also facing the same bug. Hope it's fixed soon.